packages/grid/src/vaadin-grid-scroll-mixin.js

```diff
@@ -511,11 +511,7 @@ export const ScrollMixin = (superClass) =>
         }
       });
 
-      // Only update the --_grid-horizontal-scroll-position custom property when navigating
-      // on row focus mode to avoid performance issues.
-      if (this.hasAttribute('navigating') && this.__rowFocusMode) {
-        this.$.table.style.setProperty('--_grid-horizontal-scroll-position', `${-x}px`);
-      }
+      this.$.table.style.setProperty('--_grid-horizontal-scroll-position', `${-x}px`);
```

Comment:
@tomivirkki, how big of an effect on performance this might have? A style calc and layout steps probably happen after each scroll event already since we are updating the transforms of all row elements.

Reply:
Unfortunately, it seems removing the check would re-introduce https://github.com/vaadin/flow-components/issues/2930

Currently:

https://github.com/user-attachments/assets/5e9af01c-7e91-425f-a2dc-9594f30f834a

With the check removed:

https://github.com/user-attachments/assets/2e5041ba-46a0-4993-81bf-48a18126c76f